executor/include/executor/trait/common_traits.hpp

```diff
@@ -21,4 +21,21 @@ using void_t = void;
 template <typename T>
 using remove_cv_ref_t = std::remove_cv_t<std::remove_reference_t<T>>;
 
+// In accordance with equality_comparable concept in C++ 20
+
+template <typename T1, typename T2, typename = void>
+struct equality_comparable : std::false_type {};
+
+template <typename T1, typename T2>
+struct equality_comparable<
+    T1, T2,
+    executor::void_t<decltype(std::declval<T1>() == std::declval<T2>(),
+                              std::declval<T2>() == std::declval<T1>(),
+                              std::declval<T1>() != std::declval<T2>(),
```

Comment:
This != looks suspicious.

Reply:
Fixed the issue, was missing `std::declval<T2>() != std::declval<T1>()`